app/assets/javascripts/oxalis/model/binary/data_cube.js

```diff
@@ -328,18 +328,26 @@ class DataCube {
       if (bucket instanceof DataBucket) {
         const voxelIndex = this.getVoxelIndex(voxel);
 
-        const labelFunc = (data: Uint8Array): void => {
-          // Write label in little endian order
-          for (let i = 0; i < this.BYTE_OFFSET; i++) {
-            data[voxelIndex + i] = (label >> (i * 8)) & 0xff;
-          }
-        };
-        bucket.label(labelFunc);
+        let shouldUpdateVoxel = true;
+        if (activeCellId != null) {
+          const voxelValue = this.getMappedDataValue(voxel);
```

Comment:
Is this lookup expensive? After all this function  will be called quite a lot.

Reply:
Shouldn't be too expensive for most cases.